integration-tests/java/src/integration-test/java/com/stellarstation/api/test/satellite/SatelliteStreamerTest.java

```diff
@@ -124,19 +115,83 @@ public void onCompleted() {}
             .setEnableEvents(true)
             .build());
 
-    ListenableFuture<Boolean> timeoutFuture =
-        Futures.withTimeout(
-            commandTestFuture, 30, TimeUnit.SECONDS, Executors.newSingleThreadScheduledExecutor());
+    TimeUnit.SECONDS.sleep(20);
+    requestObserver.onCompleted();
 
-    // Check safe mode state is changed..
+    ListenableFuture<Boolean> timeoutFuture = createTimeoutFuture(testCompletionFuture, 3);
     assertThat(timeoutFuture.get()).isTrue();
 
-    // Check antenna states are valid.
-    assertThat(azimuthCommand).containsOnly(1.0);
-    assertThat(azimuthMeasured).containsOnly(1.02);
-    assertThat(elevationCommand).containsOnly(20.0);
-    assertThat(elevationMeasured).containsOnly(19.5);
+    assertThat(responseObserver.getSafeModeStates()).containsExactlyInAnyOrder(0, 1);
+  }
+
+  @Test
+  void events() throws Exception {
+    final SettableFuture<Boolean> testCompletionFuture = SettableFuture.create();
+
+    class EventTestStreamObserver implements StreamObserver<SatelliteStreamResponse> {
+      private List<AntennaState> antennaStates = new ArrayList();
+
+      @Override
+      public void onNext(SatelliteStreamResponse response) {
+        if (response.hasStreamEvent()) {
+          GroundStationState gsState =
+              response.getStreamEvent().getPlanMonitoringEvent().getGroundStationState();
+          antennaStates.add(gsState.getAntenna());
+        }
+      }
+
+      @Override
+      public void onError(Throwable t) {
+        testCompletionFuture.setException(t);
+      }
+
+      @Override
+      public void onCompleted() {
+        testCompletionFuture.set(true);
+      }
+
+      public List<AntennaState> getAntennaStates() {
+        return ImmutableList.copyOf(antennaStates);
+      }
+    }
+
+    EventTestStreamObserver responseObserver = new EventTestStreamObserver();
+    StreamObserver<SatelliteStreamRequest> requestObserver = streamer.openStream(responseObserver);
+
+    requestObserver.onNext(
+        SatelliteStreamRequest.newBuilder()
+            .setSatelliteId(SATELLITE_ID)
+            .setEnableEvents(true)
+            .build());
 
+    TimeUnit.SECONDS.sleep(10);
```

Comment:
Same as the comment above. If we call onCompleted inside onNext after the first status arrives, then I think we can get rid of this sleep.